tool/jt.rb

```diff
@@ -29,6 +29,9 @@
 GRAAL_DIR = File.expand_path('../graal', TRUFFLERUBY_DIR)
 PROFILES_DIR = "#{TRUFFLERUBY_DIR}/profiles"
 
+CACHE_EXTRA_DIR = File.expand_path('~/.mx/cache/extra')
```

Comment:
I'm not sure if this could be problematic by reusing mx's folder for another purpose (at least the naming is different than other subfolders under `~/.mx/cache`). Probably it's fine.

Maybe we should try to actually use mx to download those dependencies, and to extract them?
I see there is something like that for e.g. `bzip2` (extracted in `.mx/cache/BZIP2_bf7badf7e248e0ecf465d33c2f5aeec774209227/bzip2.extracted/bzip2-1.0.8`), `ninja`, `ninja-syntax`, `LLVM_ORG*`.
OTOH we already need to know the final path to those from `jt`, so not sure there is a convenient way to find that without needing to shell out to mx (which is slow).

I'm unsure if storing the JDK (and particularly the extracted version, also applies to Eclipse) under `~/.mx/cache` is a good idea.

Notably this will explode the size of `~/.mx/cache`, which is for instance cached in [.travis.yml](https://github.com/oracle/truffleruby/blob/cb774a0c04ff1f63f95180ef335e1f173ce78152/.travis.yml). That sounds like it could actually slow down things a lot. From Travis' UI it shows it's 962MB vs 351MB before.
In https://travis-ci.org/github/oracle/truffleruby/jobs/708147947 it took 51 seconds to upload the cache, vs 5s to download it (+12s to extract it).
The main reason `~/.mx/cache` is cached in TruffleRuby's TravisCI is to avoid hitting many different and rather slow servers. This doesn't apply for the JDK and Eclipse where they are downloaded from GitHub.

Essentially having the extracted version under `.mx/cache` means doubling the size and creating many files to save/restore, which would likely be just as fast to re-extract every time.

Maybe a better way would be to only store the downloaded archive under `.mx/cache` and leave the extracted version in truffleruby's parent directory (or somewhere else). It would only partially address your clutter point though.

What's the main issue you're trying to solve here?
Isn't downloading those from GitHub about as fast as from anywhere?
Should we maybe have a `deps` or so directory under `truffleruby` and download + extract there?

Reply:
> Maybe we should try to actually use mx to download those dependencies, and to extract them?

Yeah that'd be great... can we do that? I never understood why JVMCI isn't an mx dependency like all others. Last time I asked I think they said it's because people use so many difference JDKs?

> Notably this will explode the size of `~/.mx/cache`

It makes the cache larger yeah... because it's caching more. Yes having both the tarball and expanded version is redundant - maybe we should delete the tarball then?

> This doesn't apply for the JDK and Eclipse where they are downloaded from GitHub.

Presumably the cache is stored much closer than GitHub?

> which would likely be just as fast to re-extract every time

Seems redundant to unpack the overall cache, then unpack another tarball from within that cache? That's packing within packing.

> Maybe a better way would be to only store the downloaded archive

If anything I think store the unpacked version and delete the archive. But I think storing both is fine.

> What's the main issue you're trying to solve here?

Having to cache multiple things in multiple different ways, which is hard when they don't even have long-term predictable names that you can discover before you've got `jt` and things running.

> Isn't downloading those from GitHub about as fast as from anywhere?

Pulling out of the Microsoft cloud and into the Google cloud where we run CI every time is a waste. I guess it's also a waste to pull out of Microsoft and into wherever Travis runs.

Reply:
I think other things are cached extracted as well? For example my cache includes things like

```
chrisseaton@Chris-Seatons-MacBook-Pro ~ % ls ~/.mx/cache/LLVM_ORG_SRC_27bea70346768ee43893df3f65cf785c5b5d0342
llvm-org-src.extracted		llvm-org-src.tar.gz		llvm-org-src.tar.gz.sha1
% ls ~/.mx/cache/LLVM_ORG_SRC_27bea70346768ee43893df3f65cf785c5b5d0342/llvm-org-src.extracted
README.md		clang-tools-extra	debuginfo-tests		libcxx			libunwind		lldb			llvm			parallel-libs		pstl
clang			compiler-rt		libclc			libcxxabi		lld			llgo			openmp			polly
```

Reply:
> Yeah that'd be great... can we do that? I never understood why JVMCI isn't an mx dependency like all others. Last time I asked I think they said it's because people use so many difference JDKs?

How would we find out where mx downloaded those though? A Dir.glob and hope there is only one match seems rather fragile.

Reply:
I mean that `jt` shouldn't manage it - there should be a normal `mx` dependency on JVMCI as a suite.

But anyway, yeah I don't know how we'd do that - so the simple changes in this PR get it done instead.

Reply:
OK, let's try the approach in this PR.